### ISSUE TITLE
ALLOWED_EAN_EXTENSIONS not working

### ISSUE THREAD

**Issue by akrezic1** [journeyapps/zxing-android-embedded#225]
I need to read barcode with EAN 2 and there is a problem when I do it with IntentIntegrator and internal library call, but it works when using intent to "com.google.zxing.client.android.SCAN".

Logcat shows this message with library:
`Ignoring hint ALLOWED_EAN_EXTENSIONS because it is not assignable from [I@19d57a3`

And with intent for external app:
`Hints from the Intent: {ALLOWED_EAN_EXTENSIONS=[I@d363e9d}`

So it seems like the last "}" was somehow lost, but it doesn't make sense to me. Is anyone using this feature and does it work?

This is how I use IntentIntegrator:

```
IntentIntegrator integrator = IntentIntegrator.forFragment(PollFragment.this);
integrator.addExtra("ALLOWED_EAN_EXTENSIONS", new int[] {5, 2});
integrator.setCaptureActivity(BarcodeScannerActivity.class);
integrator.setOrientationLocked(false);
integrator.initiateScan();
```

**Comment by rkistner** [comment]
Looks like a bug in `IntentIntegrator` / `DecodeHintManager`. You can work around this by setting it directly from your `BarcodeScannerActivity`, instead of via `IntentIntegrator`.

**Comment by akrezic1** [comment]
Thanks, I'll try it out later today and reply here with result.

**Comment by akrezic1** [comment]
Ok strange thing just happened 😄 

Seems like this flag "ALLOWED_EAN_EXTENSIONS" is not even needed. In Intent that is returned to onActivityResult() you can find that value with this:

`String eanExtension = data.getStringExtra("SCAN_RESULT_UPC_EAN_EXTENSION");`

Not sure if that was intentional or what, but it works. Thanks again!

**Comment by rkistner** [comment]
The `ALLOWED_EAN_EXTENSIONS` flag is just to prevent the scanning process from completing before it found an extension. So without the flag it could still return the extension, but you're not guaranteed to get it.

**Comment by akrezic1** [comment]
Ok thanks for clearing it out.

**Comment by rkistner** [comment]
This should finally be fixed in v4.0.2.
